src/main/java/com/arpnetworking/metrics/impl/StenoLogSink.java

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright 2015 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.impl;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.encoder.Encoder;
+import com.arpnetworking.logback.StenoEncoder;
+import com.arpnetworking.logback.StenoMarker;
+import com.arpnetworking.metrics.CompoundUnit;
+import com.arpnetworking.metrics.Event;
+import com.arpnetworking.metrics.Quantity;
+import com.arpnetworking.metrics.Unit;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Implementation of <code>Sink</code> for the query log file with Steno
+ * wrapped events.
+ *
+ * @author Ville Koskela (vkoskela at groupon dot com)
+ */
+public class StenoLogSink extends BaseFileSink {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void record(final Event event) {
+
+        // CHECKSTYLE.OFF: IllegalInstantiation - No Guava dependency here.
+        final Map<String, String> filteredAnnotations = new HashMap<>(event.getAnnotations());
+        // CHECKSTYLE.ON: IllegalInstantiation
+        filteredAnnotations.remove("_host");
+        filteredAnnotations.remove("_id");
+
+        try {
+            getMetricsLogger().info(
```

Comment:
Not really. The whole point of the sink in this context is to compartmentalize the serialization to a particular format (note: 2g will split "writing" and "serialization" to sink and formatter). The Logback-Steno builder pattern delegates serialization to the appender thread using its ObjectMapper instance. This would require either 1) Ensuring the _shared_ java client classes serialize as desired which is not possible in general; or 2) Copying the _shared_ java client class instances into sink (or format) specific instances which is analogous to serializing them directly except slower with more object churn. Does this make sense?